packages/salesforcedx-vscode-core/src/commands/functions/forceFunctionStart.ts

```diff
@@ -5,40 +5,150 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 import {
+  CliCommandExecutor,
   Command,
   SfdxCommandBuilder
 } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
-import { Uri } from 'vscode';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as fs from 'fs';
+import * as path from 'path';
+import { Observable } from 'rxjs/Observable';
+import { Subject } from 'rxjs/Subject';
+import * as vscode from 'vscode';
+import { channelService } from '../../channels';
 import { nls } from '../../messages';
+import { notificationService, ProgressNotification } from '../../notifications';
+import { taskViewService } from '../../statuses';
+import { telemetryService } from '../../telemetry';
 import {
-  CommandletExecutor,
   FilePathGatherer,
   SfdxCommandlet,
   SfdxCommandletExecutor,
   SfdxWorkspaceChecker
 } from '../util';
 
-export class ForceFunctionStart extends SfdxCommandletExecutor<string> {
-  public build(sourceFsPath: string): Command {
-    this.executionCwd = sourceFsPath;
+import { Uri, window } from 'vscode';
+import { FunctionService } from './functionService';
+
+export class ForceFunctionStartExecutor extends SfdxCommandletExecutor<string> {
+  public build(functionDirPath: string): Command {
+    this.executionCwd = functionDirPath;
     return new SfdxCommandBuilder()
       .withDescription(nls.localize('force_function_start_text'))
       .withArg('evergreen:function:start')
       .withArg('--verbose')
       .withLogName('force_function_start')
       .build();
   }
+
+  /**
+   * Locate the directory that has function.toml
+   * @param sourceFsPath path to start function from
+   */
+  public static getFunctionDir(sourceFsPath: string) {
+    let current = fs.lstatSync(sourceFsPath).isDirectory()
+      ? sourceFsPath
+      : path.dirname(sourceFsPath);
+    const { root } = path.parse(sourceFsPath);
+    while (current !== root) {
+      const tomlPath = path.join(current, 'function.toml');
```

Comment:
If a customer has custom directories within function and wants to start function from there, we should still be allowing that? In that case, we need to recursively search for the function's root directory. Is this something we take care now?

Reply:
Yes, it looks up by `path.dirname` until it couldn't find it

Reply:
I'll add some more comments to the jsdocs here

Reply:
We need to change this code to stop iterating when it reaches the project's `functions` directory or the root of the project. Right now it will iterate over all the folders in the file path initially provided if it doesn't find a `function.toml`.

For example, if I trigger this from a function with no `.toml` or by right clicking the `functions` folder the path could look like this `'/Users/myuser/Documents/myDXProject/functions/littleBot'.
This code will keep iterating until reaching `Users` in the scenario described here.

Reply:
I could add this to stop at project's workspace root. We could not exclude the possibility where there is another folder named `functions` inside the function folder.

Reply:
Updated